src/implementations/sign/index.ts

```diff
@@ -1 +1,42 @@
-export * from "./sign";
+import { PrivateKeyOption } from "../../commands/shared";
+import { getPrivateKey } from "../utils/wallet";
+import { documentsInDirectory, readOpenAttestationFile, writeDocumentToDisk, isDir } from "../utils/disk";
+import { signDocument, defaultSigners } from "@govtechsg/oa-did-sign";
+import path from "path";
+import mkdirp from "mkdirp";
+
+export interface SignCommand extends PrivateKeyOption {
+  rawDocumentsPath: string;
+  outputDir: string;
+  publicKey: string;
+  algorithm: string;
+}
+
+export const supportedAlgorithms = Array.from(defaultSigners.keys());
+
+export const sign = async ({
+  rawDocumentsPath,
+  outputDir,
+  publicKey,
+  algorithm,
+  key,
+  keyFile,
+}: SignCommand): Promise<void> => {
+  const privateKey = await getPrivateKey({ key, keyFile });
+  if (!privateKey) throw new Error("Private key is not specified (use -k or -f to specify key)");
+  if (!isDir(outputDir)) throw new Error("output-dir must be a director");
+
+  // Create output dir
+  if (outputDir) {
+    mkdirp.sync(outputDir);
+  }
+
+  // Sign each document independently and save to output directory
+  const documentFileNames = await documentsInDirectory(rawDocumentsPath);
+  const deferredProcesses = documentFileNames.map(async (fileName) => {
```

Comment:
I imagine there could be problem with large directory containing a lot of files (fd limit) but let's wait and see if the problem happen

Reply:
Yep i agree. Thought of that when writing this as well but I'll wait and see. 

There is a nice lib to manage async concurrency(https://www.npmjs.com/package/async).